liquibase-core/src/main/java/liquibase/sqlgenerator/core/SetNullableGenerator.java

```diff
@@ -74,6 +74,8 @@ public ValidationErrors validate(SetNullableStatement setNullableStatement, Data
                 nullableString = "";
             }
             sql = "ALTER TABLE " + database.escapeTableName(statement.getCatalogName(), statement.getSchemaName(), statement.getTableName()) + " MODIFY (" + database.escapeColumnName(statement.getCatalogName(), statement.getSchemaName(), statement.getTableName(), statement.getColumnName()) + " " + DataTypeFactory.getInstance().fromDescription(statement.getColumnDataType(), database).toDatabaseDataType(database) + nullableString + ")";
+        } else if (database instanceof FirebirdDatabase && !(database instanceof Firebird3Database)) {
+            sql = "UPDATE RDB$RELATION_FIELDS SET RDB$NULL_FLAG = " + (statement.isNullable() ? "1" : "NULL") + " WHERE RDB$RELATION_NAME = " + database.escapeTableName(statement.getCatalogName(), statement.getSchemaName(), statement.getTableName()) + " AND RDB$FIELD_NAME = " + database.escapeColumnName(statement.getCatalogName(), statement.getSchemaName(), statement.getTableName(), statement.getColumnName());
```

Comment:
Can you explain why this is referring to system tables and not just doing an `ALTER TABLE` statement?

Reply:
Added a commit with a comment in the code, also referencing the official Firebird FAQ